### DISCUSSION TITLE
New draft issues are hidden when created within a view with filters

### DISCUSSION THREAD

**Discussion by orospakr** [community/community#8079] [Projects and Issues] [answered]
Right now, when creating an issue in a view in the project (that has any filters applied at all) then your issue will disappear and you will see this disappointing prompt:

![image](https://user-images.githubusercontent.com/16714/143284929-0b1860e2-1837-4cbb-9bd4-c2ca4f5b8064.png)

## Use Case

I'm working on shaping/planning work either on my own or with my team on a video call/projector, and want to draft up new issues right there in context established by the filter rules on my view. Ie, I have a view tab for a milestone.

## Proposed Solution

Give draft issues the same fields as real issues: milestone, labels, etc. (or otherwise somehow associate them with the View itself in some other way). When creating a draft issue in a view, where possible, set those properties values on the draft issue so it will match the view's filters. Ie., if the view is filtering for a milestone "2.0", then set the milestone property on the issue to "2.0".

Then, when converting a draft issue to real issues, bring over the properties of the draft issue (or containing view) so that the newly created issue will match the filter and not disappear.

Thank you!

**Comment by jasononeil** [comment, marked as answer]
Apparently this is "coming next week" 🤞 

https://github.com/github/feedback/discussions/6803#discussioncomment-1699204

**Reply by orospakr** [reply]
oho, awesome!

**Reply by azenMatt** [reply]
@jasononeil @orospakr - I'm glad we could deliver on this one before the end of the week! 

https://github.blog/changelog/2021-12-02-the-new-github-issues-12-02-update/

Check it out and let me know if it helps! 

Two limitations to be aware of:
- If it isn't clear what value should be added, (for example you have an OR filter) the old behaviour is used
- If you add a draft and a filter is over assignee, milestone, repo - then we can't add those yet, so the old behaviour is used. 

Assignees are coming to drafts soon (Maybe by the end of the year... 🤞) and then others will follow!

**Reply by orospakr** [reply]
Hey @azenMatt thank you for this!

~However I do notice that *draft* issues still have the disappearing issue. So if you don't use the `repo:` attribute to skip the draft phase, your issue will still disappear.~

Oh, I'm sorry, I lack reading comprehension this early in the morning. Coming later in the year hopefully. Got it :)

Thanks again

**Comment by jasononeil** [comment]
Thanks! At first I thought it wasn’t working for me, but then I realised if I simplified the filtering on the views it starts working.

Do you have any advice / docs on how it prefills values based on current view filters? I’d be interested to know which kind of filters prevent it. I can’t remember if it was negative filters, multiple filters, filtering on multiple values, or a combination.

Either way - thanks! This change has improved the feature a lot for me

**Reply by azenMatt** [reply]
No docs on this yet, but the general approach is that we 'do it if we can' 

That means we don't if either:

1) It isn't obvious what value should be added (for example there is an OR query)
2) The value can't be added due to permissions on the target item. (for example, if a user doesn't have access to repo A we can't assign them to an issue from that repo)

**Reply by azenMatt** [reply]
Open to ideas on how we can make this work best for more complicated queries! We do want to keep iterating on this!
